pkg/patch/patch.go

```diff
@@ -23,7 +23,11 @@ func CreatePatch(oldJSON []byte, newObj interface{}, response *v1.AdmissionRespo
 	if err != nil {
 		return fmt.Errorf("failed to marshal generated patch to JSON: %w", err)
 	}
-
+	// json.Marshal() function treats an empty slice as "[]" and a nil pointer as "null" when performing the marshal,
```

Comment:
I think it would be cleaner/more efficient to just ensure `patch.Patches` is non-nil and non-empty slice by returning nil if `len(patch.Patches) == 0` before even marshalling it and checking string responses.

Reply:
Thanks for this good idea! 
Code is updated.